### PR TITLE
add solution of second part

### PR DESCRIPTION
**Pull request by ssaviak**
[DEMO LINK](https://ssaviak.github.io/react_todo-app-add-and-delete/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,58 @@
...
+          <TodoList />
+        </section>
+
+        {/* Hide the footer if there are no todos */}
+        {todos.length !== 0 && (<Footer />)}
+      </div>
+
+      {/* Notification is shown in case of any error */}
+      {/* Add the 'hidden' class to hide the message smoothly */}
+      <Error />
```
Comment (olya-shyian) [inline src/App.tsx:55]: use check

`{errorMessage && <Error />}`

File: src/components/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    setErrorMessage('');
+
+    const tempTodo: Todo = {
+      title: query.trim(),
+      completed: false,
+      id: 0,
+      userId: USER_ID,
+    };
+
+    if (query.trim().length === 0) {
```
Comment (olya-shyian) [inline src/components/Header.tsx:39]: instead write 
`(!query.trim().length)`

File: src/App.tsx [hunk 3]
```diff
@@ -1,24 +1,58 @@
...
+  } = useContext(TodoContext);
+
+  const handleOverLoad = () => {
+    setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  };
+
+  useEffect(() => {
+    getTodos(84)
```
Comment (olya-shyian) [inline src/App.tsx:28]: please do updates according to comments from previous task

https://github.com/mate-academy/react_todo-app-loading-todos/pull/1115#pullrequestreview-1862804333

**Review by olya-shyian — CHANGES_REQUESTED** [review]
Goood!✨

Also the loader should be each time you add or delete Todos (and in other cases)

Check the working version
image

File: src/contexts/TodoContext.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    filteredTodos: todos.filter(todo => {
+      switch (filters.status) {
+        case 'uncompleted':
+          return !todo.completed;
+        case 'completed':
+          return todo.completed;
+        default:
+          return true;
+      }
+    }),
```
Comment (DanilWeda) [inline src/contexts/TodoContext.tsx:55]: can move it outside

File: src/App.tsx [hunk 5]
```diff
@@ -1,24 +1,58 @@
...
+      .then((res) => setTodos(res))
```
Comment (DanilWeda) [inline src/App.tsx:29]: or you can write 
```suggestion
      .then(setTodos)
```

**Review by DanilWeda — CHANGES_REQUESTED** [review]
Great work! 
Check comments about small issues.
Many thanks!

**Review by olya-shyian — APPROVED** [review]
Thank you!🌟

**Pull request closed without merging**